Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -96,12 +96,12 @@ WarpXParticleContainer::AllocData ()
 }
 
 void
-WarpXParticleContainer::AddNParticles (int lev,
+WarpXParticleContainer::AddNParticles (int /*lev*/,
                                        int n, const ParticleReal* x, const ParticleReal* y, const ParticleReal* z,
                                        const ParticleReal* vx, const ParticleReal* vy, const ParticleReal* vz,
                                        int nattr, const ParticleReal* attr, int uniqueparticles, int id)
 {
-    BL_ASSERT(nattr == 1);
+    BL_ASSERT(nattr == 1); //! @fixme nattr is unused below: false sense of safety
```

Comment:
I am trying to understand this line. Is it an amrex macro for when the `debug` flag is active?

Reply:
This line is a macro that [expands to a runtime test](https://github.com/AMReX-Codes/amrex/blob/e7d931ce8668d7b770af4adf3be6b2b179df3e1b/Src/Base/AMReX_BLassert.H) if `nattr` is of value `1`.
If not, the code will throw an exception if any other value is passed.

Now the problem with this line is the following: `nattr` is not used *elsewhere* in the function body. That means nothing is actually depending on it and the assert as well as asking for `nattr` in the interface does not change/validate anything.

Reply:
Thank you @ax3l !! It makes sense now (the == works like in an if statement).
Then it could be removed I suppose.